src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -0,0 +1,602 @@
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.compute.OperatingSystemTypes;
+import com.microsoft.azure.management.compute.VirtualMachine;
+import com.microsoft.azure.management.compute.VirtualMachineCustomImage;
+import com.microsoft.azure.management.compute.VirtualMachineDataDisk;
+import com.microsoft.azure.management.compute.VirtualMachineExtension;
+import com.microsoft.azure.management.compute.VirtualMachineSizeTypes;
+import com.microsoft.azure.management.network.Network;
+import com.microsoft.azure.management.network.NetworkInterface;
+import com.microsoft.azure.management.network.NetworkSecurityGroup;
+import com.microsoft.azure.management.network.PublicIpAddress;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.utils.SdkContext;
+
+import lombok.Getter;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 01/03/17
+ */
+@Component
+public class AzureProvider implements CloudProvider {
+
+    private final Logger logger = Logger.getLogger(AzureProvider.class);
+
+    @Getter
+    private final String type = "azure";
+
+    private static final Region defaultRegion = Region.US_EAST;
+
+    private static final VirtualMachineSizeTypes defaultVMSize = VirtualMachineSizeTypes.STANDARD_D1_V2;
+
+    private static final int resourcesNameExtraChars = 5;
+
+    private static final String virtualNetworkNameBase = "vnet";
+
+    private static final String publicIPAddressNameBase = "ip";
+
+    private static final String networkSecurityGroupNameBase = "sg";
+
+    private static final String networkInterfaceNameBase = "if";
+
+    private static final String osDiskNameBase = "os";
+
+    private static final String defaultUsername = "activeeon";
+
+    private static final String defaultPassword = "Act1v€0N";
+
+    private static final String defaultPrivateNetworkCIDR = "10.0.0.0/24";
+
+    private static final Boolean defaultStaticPublicIP = true;
+
+    private static final String scriptExtensionPublisher = "Microsoft.Azure.Extensions";
+
+    private static final String scriptExtensionType = "CustomScript";
+
+    private static final String scriptExtensionVersion = "2.0";
+
+    private static final String scriptExtensionCmdKey = "commandToExecute";
+
+    @Autowired
+    private AzureServiceCache azureServiceCache;
+
+    @Autowired
+    private AzureProviderUtils azureProviderUtils;
+
+    @Override
+    public Set<Instance> createInstance(Infrastructure infrastructure, Instance instance) {
+
+        Azure azureService = azureServiceCache.getService(infrastructure);
+        String instanceTag = Optional.ofNullable(instance.getTag())
+                                     .orElseThrow(() -> new RuntimeException("ERROR missing instance tag/name from instance: '" +
```

Comment:
Good question, this could be done in the entire project. I simply re-used the same convention than existing code but I will open an issue about this as a quality improvement.